### PR TITLE
Added missing OUTPUT_DIR configs in the script

### PR DESCRIPTION
**Pull request by ezio-auditore**
The script wasn't working out of the box without running `make bundle` at least once. This change fixes that now.

### CODE REVIEW

**Comment by ezio-auditore** [comment]
@renan-campos would be great to have your review on this

File: Makefile [hunk 1]
```diff
@@ -164,8 +166,8 @@ bundle: manifests kustomize
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) $(BUNDLE_FLAGS)
-	cp config/metadata/* bundle/metadata/
-	operator-sdk bundle validate ./bundle
+	cp config/metadata/* $(OUTPUT_DIR)/metadata/
```
Comment (renan-campos) [inline Makefile:169]: Oh I see! This directory was getting copied because the dependencies.yaml is located there. Good catch :)

**Review by renan-campos — APPROVED** [review]
LGTM!

File: tools/deploy-with-olm.sh [hunk 2]
```diff
@@ -41,19 +41,20 @@ TARGET_NAMESPACE=${TARGET_NAMESPACE:-openshift-storage}
...
 
+OUTPUT_DIR=./tools/bundle
 # Generate the olm bundle image
 blue "Generate and push the olm bundle image: ${BUNDLE_IMAGE}"
-exit_on_err make bundle IMG=${DEPLOYER_IMAGE} OUTPUT_DIR=./tools/bundle
-exit_on_err sed -i "s|- name: ADDON_NAME|- {name: ADDON_NAME, value: ${ADDON_NAME}}|" ${BUNDLE_FILE}
-exit_on_err sed -i "s|- name: SOP_ENDPOINT|- {name: SOP_ENDPOINT, value: \'${SOP_ENDPOINT}\'}|" ${BUNDLE_FILE}
+exit_on_err make bundle IMG=${DEPLOYER_IMAGE} OUTPUT_DIR=${OUTPUT_DIR}
+exit_on_err sed -i "s|- name: ADDON_NAME|- {name: ADDON_NAME, value: ${ADDON_NAME}}|" ${OUTPUT_DIR}/${BUNDLE_FILE}
+exit_on_err sed -i "s|- name: SOP_ENDPOINT|- {name: SOP_ENDPOINT, value: \'${SOP_ENDPOINT}\'}|" ${OUTPUT_DIR}/${BUNDLE_FILE}
```
Comment (nb-ohad) [inline tools/deploy-with-olm.sh:57]: Why do we need to `BUNDLE_FILE` separately then `OUTPUT_DIR`? we never use it on its own:

```suggestion
OUTPUT_DIR=./tools/bundle
BUNDLE_FILE=${OUTPUT_DIR}/manifests/ocs-osd-deployer.clusterserviceversion.yaml
CLUSTER_SIZE=${CLUSTER_SIZE:-1}

# Generate the deployer image 
blue "Generate deployer image: ${DEPLOYER_IMAGE}"
exit_on_err make docker-build IMG=${DEPLOYER_IMAGE}
exit_on_err make docker-push IMG=${DEPLOYER_IMAGE}

# Generate the olm bundle image
blue "Generate and push the olm bundle image: ${BUNDLE_IMAGE}"
exit_on_err make bundle IMG=${DEPLOYER_IMAGE} OUTPUT_DIR=${OUTPUT_DIR}
exit_on_err sed -i "s|- name: ADDON_NAME|- {name: ADDON_NAME, value: ${ADDON_NAME}}|" ${BUNDLE_FILE}
exit_on_err sed -i "s|- name: SOP_ENDPOINT|- {name: SOP_ENDPOINT, value: \'${SOP_ENDPOINT}\'}|" ${BUNDLE_FILE}
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: tools/deploy-with-olm.sh (hunk 2 above)
Comment (ezio-auditore) [reply tools/deploy-with-olm.sh]: Yes, thanks for pointing it out.Looks cleaner.Fixed in next update

File: Makefile [hunk 3]
```diff
@@ -14,6 +14,8 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 
 ifneq ($(origin OUTPUT_DIR), undefined)
 BUNDLE_OUTPUT_DIR = --output-dir=$(OUTPUT_DIR)
+else
+OUTPUT_DIR = bundle
 endif
```
Comment (nb-ohad) [inline Makefile:19]: There is a syntax for setting a value if `undefined`: `?=`
Using it you can re-write this section like this:

```suggestion
OUTPUT_DIR ?= bundle
BUNDLE_FLAGS = --output-dir=$(OUTPUT_DIR)
```

File: tools/deploy-with-olm.sh [hunk 4]
```diff
@@ -41,17 +41,18 @@ TARGET_NAMESPACE=${TARGET_NAMESPACE:-openshift-storage}
...
-BUNDLE_FILE=bundle/manifests/ocs-osd-deployer.clusterserviceversion.yaml
+BUNDLE_FILE=${OUTPUT_DIR}/manifests/ocs-osd-deployer.clusterserviceversion.yaml
 CLUSTER_SIZE=${CLUSTER_SIZE:-1}
 
 # Generate the deployer image 
 blue "Generate deployer image: ${DEPLOYER_IMAGE}"
 exit_on_err make docker-build IMG=${DEPLOYER_IMAGE}
 exit_on_err make docker-push IMG=${DEPLOYER_IMAGE}
 
+OUTPUT_DIR=./tools/bundle
```
Comment (nb-ohad) [inline tools/deploy-with-olm.sh:52]: You are defining this value after you already use it (line 44). Please move this immediately before line 44

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**